### DISCUSSION TITLE
Flaky data, on boot putting inverter/charger into eqaulize mode

### DISCUSSION THREAD

**Discussion by jonfortier** [CharlesGodwin/pymagnum#59] [General]
Hey all.

I have pymagnum running on a pi b+. I tried the first adapter in the instructions but could not get it to see any data. I then bought a modular one, but I was getting results like 140V incoming, or 8 volt, all random. I then say Charles noted that he tried the one I purchased and had bad data problems. I bought a CAN HAT and have it running. First thing, it wiged out my inverter on first boot. I reversed polarity and that helped. But when I boot the pi now, it puts my inverter/charger into equalize mode. If I turn that off with the panel, MQTT reports the inverter mode is 84, although the display says float charging. I am also getting erroneous data, like BMK battery voltage of 75 (they are 12v), inverter battery voltage of 331, and an incoming ac voltage of 8. note, I have a separate meter that say the voltage is OK.

Any thought on the CAN HAT data issue?

Absent that, I think I'm going to go back to trying a usb to rs485. Does the Waveshare industrial work?

Thanks,

**Comment by CharlesGodwin** [comment]
I am perplexed. I've never seen this. Just guessing but ...... Are your RJ11 wires right. If the wrong wires were attached, i don't know. I added code in my stuff to wait 30 seconds after boot up before attempting to connect to the serial port as I had seen some issues a long time ago. the delay worked for me.

Anything else? Are you plugged into the right port as described in my documentation?

**Reply by jonfortier** [reply]
I believe it's right. I used an RJ45 jack because I have them. pins 3 and 6 went to A and B on the CAN HAT. I'm connected to my BMK with a splitter. I also tried a "sleep 30" in rc.local before the command to start mgttlogger.

I went back to the USB RS485 as the CAN HAT was unusable, and I didn't trust it to mess up the system while I was away. Also, I have automations based on certain values. Almost all the data was wrong all the time. Things like model, errors, voltages, etc. At least with the USB, most of the data is correct, it just occasional dirty data, like 8 volts incoming, or 140. I posted this in the midnite solar post where you announced the program. However, I realized I inadvertently purchased the interface you mentioned that you were getting dirty data with, so I tried the HAT.

One thing that is weird, is that when I went back to the USB, the BMK is reporting high voltages.

here is the MQTT data. (note the high battery voltage and max battery volts, the inverter reports the bat voltage correctly)

{"datetime":"2024-02-23T18:32:47-06:00","comm_device":"/dev/ttyUSB0","device":"BMK","data":{"revision":"1.0","soc":100,"vdc":74.68,"adc":3.4,"vmin":11.23,"vmax":270.99,"amph":115,"amphtrip":6553.5,"amphout":10600,"Fault":1,"Fault_Text":"Normal"}}

**Reply by jonfortier** [reply]
Also, I tried the inerface shown in your instructions, the one with the green plug, but got no data. I think it's bad. I looked at the WaveShare industrial interface. Do you know if that works with a Pi?

**Comment by CharlesGodwin** [comment]
> I used an RJ45 jack because I have them. pins 3 and 6 went to A and B on the CAN HAT.

My [documentation ](https://pymagnum.readthedocs.io/en/latest/building_an_adaptor.html#section-10:~:text=The%20Jack%20adaptor%20is%20designed%20to%20work%20with%204%2Dpin%20or%206%2Dpin%20phone%20cords%20so%20when%20using%20this%20jack%20device%20use%20terminals%20numbered%202%20and%205.) says to use pins 2 and 5. Have you tried that?

**Reply by jonfortier** [reply]
On an RJ45 (4 pair, 8 wire) pins 3 and 6 are the same as pins 2 and 5 on an RJ11.

**Reply by CharlesGodwin** [reply]
You're right, I missed the RJ45 reference. I assumed you were using standard 4 wire (AKA RJ11) wiring. Back to my original response. I have no idea why its going wrong.
